addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxWsClient.java

```diff
@@ -255,22 +260,22 @@ boolean connect() {
 
         synchronized (state) {
             socket = new LxWebSocket();
-            wsClient = new WebSocketClient();
+            WebSocketClient client = new WebSocketClient();
+            wsClient = client;
```

Comment:
Why do you create a local variable at all?

Reply:
This is to avoid a warning about potential null access, because theoretically the indirect reference may have changed between the check and use.

‘’’
if (a.b != null) {
    x(a.b)
    y(a.b)
}
‘’’

Apparently in constructions like that compiler will not optimize to cache the referenced value, because this is not what we ask for coding it like that. If the code flow allows for preemption to another thread, the subsequent use of a.b may find b as null. This is my explanation as to why there is a warning. 
This leads to awkward constructions with local variables. I would like to eliminate that and this may lead to warning level tuning but I don’t remember now which one was responsible for that.

Reply:
If you just introduced the local variable to satisfy the warnings of the eclipse nullchecker, please undo your changes. We definitely do not want to code around those issue.

But as it is within a `synchronized` block I am assuming that `wsClient` can indeed be changed by another thread, is that correct? If not, please undo your changes as I mentioned above.